Marlin/src/gcode/feature/pause/M600.cpp

```diff
@@ -43,73 +48,154 @@
  *  L[distance] - Extrude distance for insertion (positive value) (manual reload)
  *  B[count]    - Number of times to beep, -1 for indefinite (if equipped with a buzzer)
  *
+ * Spool_Swap : Tool change after filament runout
+ *
+ *  T[int]: 0=disable : 1/2/3 Enable the max number of swapping
+ *
  *  Default values are used for omitted arguments.
  *
  */
 void GcodeSuite::M600() {
-
-  #if ENABLED(HOME_BEFORE_FILAMENT_CHANGE)
+	
+  #if ENABLED(PAUSE_SPOOL_SWAP) 
+    //extern bool swap_spool_enabled;
+	//extern int16_t swap_spool_stop;
+	const int swap_spool_tmp = parser.intval('T',(FIL_RUNOUT_SENSORS));             
+    
+	// M600 T0/T1/2/3/4 Enable/Disable     
+	if (swap_spool_tmp<FIL_RUNOUT_SENSORS){
+	  if (swap_spool_tmp>0){
+	    // Activate or deactivate if next extruder available
+	    swap_spool_enabled = true ;
+		//Apply max number of swapping 
+		swap_spool_stop = swap_spool_tmp;
+	  }
+	  // Deactivate in all case if no next extruders available
+	  else swap_spool_enabled=false; 
+	  // Applying number or next extruders
+	  
+	  return ;  
+    }      
+  #endif
+	
+  #if ENABLED(PAUSE_HOME_BEFORE)
     // Don't allow filament change without homing first
     if (axis_unhomed_error()) home_all_axes();
   #endif
 
   // Initial retract before move to filament change position
   const float retract = parser.seen('E') ? parser.value_axis_units(E_AXIS) : 0
-    #ifdef PAUSE_PARK_RETRACT_LENGTH
-      - (PAUSE_PARK_RETRACT_LENGTH)
+    #ifdef PAUSE_RET
+      - (PAUSE_RET)
     #endif
   ;
 
   // Lift Z axis
   const float z_lift = parser.linearval('Z', 0
-    #ifdef PAUSE_PARK_Z_ADD
-      + PAUSE_PARK_Z_ADD
+    #ifdef PAUSE_Z_ADD
+      + PAUSE_Z_ADD
     #endif
   );
 
   // Move XY axes to filament exchange position
   const float x_pos = parser.linearval('X', 0
-    #ifdef PAUSE_PARK_X_POS
-      + PAUSE_PARK_X_POS
+    #ifdef PAUSE_X_POS
+      + PAUSE_X_POS
+	#else
+      + current_position[X_AXIS]
     #endif
```

Comment:
Here a GOOD OPTION
Now , if YPOS not defined , the carriage move only one axis
Example , i have a moving bed , and my purge bucket is on the right 
bed don't need to move
Only the x axis
NOW ITS POSSIBLE !!
Just uncomment the definition , and YPOS = CURRENT Y POS , only x movement